ssl/ssl_ciph.c

```diff
@@ -493,8 +493,51 @@ int ssl_cipher_get_evp_md_mac(SSL_CTX *ctx, const SSL_CIPHER *sslc,
     return 1;
 }
 
+int ssl_cipher_get_evp_cipher_sn(SSL_CTX *ctx, const SSL_CIPHER *sslc,
+                                 const EVP_CIPHER **enc, size_t *inputoffs)
+{
+    int i = ssl_cipher_info_lookup(ssl_cipher_table_cipher, sslc->algorithm_enc);
+
+    if (i == -1) {
+        *enc = NULL;
+    } else {
+        if (i == SSL_ENC_NULL_IDX) {
+            /*
+             * We assume we don't care about this coming from an ENGINE so
+             * just do a normal EVP_CIPHER_fetch instead of
+             * ssl_evp_cipher_fetch()
+             */
+            *enc = EVP_CIPHER_fetch(ctx->libctx, "NULL", ctx->propq);
+        } else {
+            char *ecb_name = NULL;
+
+            if ((sslc->algorithm_enc & SSL_AES128_ANY) != 0) {
+                ecb_name = "AES-128-ECB";
+                *inputoffs = 0;
+            } else if ((sslc->algorithm_enc & SSL_AES256_ANY) != 0) {
+                ecb_name = "AES-256-ECB";
+                *inputoffs = 0;
+            } else if (ossl_assert((sslc->algorithm_enc & SSL_CHACHA20) != 0)) {
+                ecb_name = "ChaCha20";
+                *inputoffs = 4;
+            }
+
+            if (ecb_name != NULL)
+                *enc = EVP_CIPHER_fetch(ctx->libctx, ecb_name, ctx->propq);
```

Comment:
Actually the comment above made me think - should we use ssl_evp_cipher_fetch() here instead? Basically it means using NIDs instead of ecb_name.

Reply:
This would mean looking for engine implementation, before checking for a provider implementation, but with engines largely deprecated, and rarely providing better implementations of basic symmetric crypto primitives than modern CPUs, I am sceptical this is warranted. But if that's what is done for the bulk record crypto, we should do the same perhaps for sequence numbers, and change both in sync as appropriate.

Reply:
Yes, IMO these should be aligned. We will drop the engine support sooner than later but until that time it makes sense to me to keep these in sync.

Reply:
Done. Thanks for the suggestion!